packages/Select/src/Select.js

```diff
@@ -1,94 +1,139 @@
 import React from "react";
-import classNames from "classnames";
 import PropTypes from "prop-types";
+import { callAll, extractChildrenProps } from "@paprika/helpers";
 import * as constants from "@paprika/constants/lib/Constants";
+import SelectPropsCollector from "./SelectPropsCollector";
 import * as sc from "./Select.styles";
 
 const Select = React.forwardRef((props, ref) => {
-  const renderPlaceholder = () => {
-    const { placeholder } = props;
-    if (!placeholder) return null;
-    return (
-      <option disabled value="" key="placeholder">
-        {placeholder}
-      </option>
-    );
-  };
-
   const {
     a11yText,
-    className,
     children,
+    defaultValue,
     hasError,
     isDisabled,
     isReadOnly,
-    size,
+    onChange,
     placeholder,
+    size,
     value,
     ...moreProps
   } = props;
 
-  if (a11yText || placeholder) {
-    moreProps["aria-label"] = a11yText || placeholder;
+  const isControlled = value !== undefined;
+  const containerProps = extractChildrenProps(children, SelectPropsCollector);
+  const [hasNoValue, setHasNoValue] = React.useState(isControlled ? !value : !defaultValue);
+
+  const _refSelect = React.useRef();
+  const refSelect = ref || _refSelect;
+
+  const styleProps = {
+    hasError,
+    isDisabled,
+    isPlaceHolderSelected: !!placeholder && hasNoValue,
+    isReadOnly,
+    size,
+  };
+
+  function handleChange(event) {
+    const newValue = event.target.value;
+    if (hasNoValue && newValue !== "") {
+      setHasNoValue(false);
+    }
+    if (!hasNoValue && newValue === "") {
+      setHasNoValue(true);
+    }
   }
 
-  const rootClasses = classNames(
-    "form-select",
-    `form-select--${size}`,
-    { "form-select--placeholder": !value && placeholder },
-    { "form-select--is-disabled": isDisabled },
-    { "form-select--is-readonly": isReadOnly },
-    { "form-select--has-error": hasError },
-    className
-  );
+  const renderPlaceholder = () => {
+    if (!placeholder) return null;
+    return (
+      <option disabled value="" data-pka-anchor="select.placeholder">
+        {placeholder}
+      </option>
+    );
+  };
+
+  function getDefaultValue() {
+    if (isControlled) return undefined;
+    if (defaultValue === null && !!placeholder) return "";
+    return defaultValue;
+  }
```

Comment:
This ensures that the `disabled` placeholder item gets selected when no `defaultValue` is provided on an uncontrolled `<Select>`.